### ISSUE TITLE
Need for more flexible Loss Function

### ISSUE THREAD

**Issue by ashutosh1919** [tensorflow/tensorflow#36369]
**System information**
- TensorFlow version (you are using): 1.15
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Since, we know that sometimes in the training process, we need the **MAE and MSE type of behavior** simultaneously. We have this kind of loss function already - **Huber Loss**.  But in some cases, we don't need just one value, instead, we need the interval. We need the loss function which can output two values corresponding to lower and upper bounds of the interval. 
This type of loss function is called **Quantile Huber Loss**, which is taken from [this paper](https://arxiv.org/pdf/1402.4624.pdf). The behavior is shown in below image.
![73123322-db0dcf80-3fb4-11ea-852d-9dc672bd0906](https://user-images.githubusercontent.com/20843596/73537063-830c1880-444d-11ea-890c-3ed96aa98a51.png)

**Will this change the current api? How?**
No, it will not change the current api. Instead, it will just add one loss function and will include it's tests.

**Who will benefit with this feature?**
This will be very beneficial for the forecasting and time series. Especially, when the variation is very dynamic, then this loss function helps very much in comparison to existing losses in tf.

**Comment by pavithrasv** [comment]
Please feel free to send a PR to the Tensorflow Addons repository for this Loss.

**Comment by karmel** [comment]
cc @seanpmorgan for Addons

https://github.com/tensorflow/addons
